crates/gpui/src/color.rs

```diff
@@ -298,6 +291,16 @@ pub fn yellow() -> Hsla {
 }
 
 impl Hsla {
+    /// Construct an [`Hsla`] object from plain values
+    pub fn new(h: f32, s: f32, l: f32, a: f32) -> Self {
```

Comment:
Thank you @SomeoneToIgnore for your detailed response, obviously I understand all your arguments and it is true that part of this PR referred to basic formatting and we can revert this part. But for me this modification was still a little different:
- Before PR, the construction of an instance of the `Hsla` structure was done outside the structure itself, which requires in most files to import the `Hsla` structure itself for manipulations plus the function `hsla` which is outside the structure separately, rather than having a single import of `Hsla` and all the methods inside.
- Moreover it seems to me that you used the formalism that I proposed with the `new` method for the initialization of `StoryColor`, so it would be relevant to do the same thing here, no?

https://github.com/zed-industries/zed/blob/8da5e57d6d8f11ee47f1d95503648888fa803f1a/crates/story/src/story.rs#L35-L45

- There are also all the `black`, `transparent_black`, `white`, `red`, `blue`, `green`, `yellow` methods which are defined outside the `Hsla` structure and most time recalled inside its implementation: https://github.com/zed-industries/zed/blob/8da5e57d6d8f11ee47f1d95503648888fa803f1a/crates/gpui/src/color.rs#L230-L298

wouldn't it be better to put all these methods in the `Hsla` structure itself for the same reasons as those mentioned above?

For my part, I don't consider it styling but maybe I'm wrong, you have much better experience than me on the subject. Besides, congratulations for your work, I'm starting to adopt your editor little by little and I love it :)